### PR TITLE
Add a couple of assertions.

### PR DESCRIPTION
**Pull request by bangerth**
I realize that we never check that the targets for the consensus algorithms are actually valid. This patch adds assertions.

/rebuild

### CODE REVIEW

**Comment by bangerth** [comment]
Looks like I accidentally also included a couple of documentation improvements. I could break those out into a separate patch, but it might not be worth the effort...

File: include/deal.II/base/mpi_consensus_algorithms.h [hunk 1]
```diff
@@ -407,10 +407,12 @@ namespace Utilities
        *   i.e. when this process can stop waiting for requests, no IBarrier is
        *   needed.
        *
-       * @note The function
-       *   Utilities::MPI::compute_point_to_point_communication_pattern() is
-       *   used to determine the source processes, which implements a
-       *   PEX-algorithm from @cite hoefler2010scalable.
+       * @note Under the hood, this function uses
+       *   Utilities::MPI::compute_point_to_point_communication_pattern()
+       *   determine the source processes, which itself is based on the
```
Comment (kronbichler) [inline include/deal.II/base/mpi_consensus_algorithms.h:412]:
```suggestion
       *   to determine the source processes, which itself is based on the
```

File: include/deal.II/base/mpi_consensus_algorithms.h [hunk 2]
```diff
@@ -504,6 +506,7 @@ namespace Utilities
         clean_up_and_end_communication();
       };
 
+
```
Comment (kronbichler) [inline include/deal.II/base/mpi_consensus_algorithms.h:509]: While here, shouldn't we have three empty lines here?

**Review by kronbichler — APPROVED** [review]

**Comment by bangerth** [comment]
So fixed.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**